Editorial review: Document scroll-triggered animations (#43158)
mdn/content: files/en-us/web/css/reference/properties/timeline-trigger-activation-range-start/index.md

## Syntax

```css
/* Keyword */
timeline-trigger-activation-range-start: normal;

/* <length-percentage> */
timeline-trigger-activation-range-start: 20%;
timeline-trigger-activation-range-start: 350px;

/* Named timeline range */
timeline-trigger-activation-range-start: cover;
timeline-trigger-activation-range-start: exit;

/* Named timeline range plus offset */
timeline-trigger-activation-range-start: entry 40%;
timeline-trigger-activation-range-start: contain 200px;

/* Multiple range start values */
timeline-trigger-activation-range-start:
  contain,
  entry 50%;

/* Global values */
timeline-trigger-activation-range-start: inherit;
timeline-trigger-activation-range-start: initial;
timeline-trigger-activation-range-start: revert;
timeline-trigger-activation-range-start: revert-layer;
timeline-trigger-activation-range-start: unset;
```

The `timeline-trigger-activation-range-start` property is specified as one or more values, separated by commas.

### Values

- `normal`

>   - : The default value. Equivalent to `cover 0%` for a [`view()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/view) {{cssxref("timeline-trigger-source")}}, and `0%` for a [`scroll()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/scroll) `timeline-trigger-source`.

Reviewer: ```suggestion
  - : The default value. Equivalent to `cover 0%` for a [`view()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/view) {{cssxref("timeline-trigger-source")}}, and `scroll 0%` for a [`scroll()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/scroll) `timeline-trigger-source`.
```

Author: This is a new thing, but yeah, I think it makes sense to express this in terms of the `scroll` range.

Author: updated on all pages.

- {{cssxref("length-percentage")}}

>   - : Specifies a length or percentage value measured from the beginning of the timeline.

Reviewer: ```suggestion
  - : Specifies a length or percentage value measured from the beginning of the `normal` timeline.
```

I still think it's easier to put the percent relative to the normal here rather than combining the two at the end, since here it is always relative to the normal range and in the other it's always relative to the named range.

Author: Yeah, OK, updated in all places.

- {{cssxref("timeline-range-name")}}

>   - : Specifies a named timeline range within the overall timeline.

Reviewer: ```suggestion
  - : Specifies  the `0%` of the named timeline range.
```
when only one value is set, the other defaults. So this defaults to 0 and the length-percent only defaults to normal

Author: Makes sense. I've specified the appropriate equivalent on all pages, with one small tweak for grammar. The structure is now:

> Specifies the start (`0%`) of the named timeline range.

- `<timeline-range-name>` `<length-percentage>`
  - : Specifies a length or percentage value measured from the beginning of the specified named timeline range.

> Percentages are relative to the length of the named timeline range if one is specified, or the entire timeline if not.

Reviewer: "or the entire timeline if not." needs to be explained 
```suggestion
Percentages are relative to the length of the named timeline range if one is specified, or the entire `normal` timeline if not.
```
these changes need to be made on all pages.

Author: I've updated this wording on all pages to:

> , or the timeline represented by `normal` if not.

I think this is a bit more intuitive, given that what normal equates to is explained in the longhand syntax sections.

## Description

When creating [CSS scroll-triggered animations](/en-US/docs/Web/CSS/Guides/Animation_triggers/Using_scroll-triggered_animations), the `timeline-trigger-activation-range-start` property can be used to explicitly define the start of the trigger's [activation range](/en-US/docs/Web/CSS/Reference/Properties/timeline-trigger-activation-range#description).

Allowed values for the `timeline-trigger-activation-range-start` property are:

- `normal`
- A `<length-percentage>`
- A `<timeline-range-name>`

> - A `<timeline-range-name>` and a `<length-percentage>` separated by a space.

Reviewer: ```suggestion
```
not nessary (that's the values section)

Author: Good call. I've nixed this on all pages.

> The default value, `normal`, is equivalent to `cover 0%` for a [`view()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/view) {{cssxref("timeline-trigger-source")}}, and `0%` for a [`scroll()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/scroll) `timeline-trigger-source`.

Reviewer: what about ident as a source?

Author: Ah, this is a really good point; I need to generalize a bit more ;-)

I've updated the language in all places to:

> Equivalent to `cover 0%` for a [view progress timeline](/en-US/docs/Web/CSS/Guides/Scroll-driven_animations/Timelines#view_progress_timelines) {{cssxref("timeline-trigger-source")}}, and `scroll 0%` for a [scroll progress timeline](/en-US/docs/Web/CSS/Guides/Scroll-driven_animations/Timelines#scroll_progress_timelines) `timeline-trigger-source`.

> If the `<timeline-range-name>` value does not include a `<length-percentage>`, it defaults to `0%` of the named timeline range. If a `<timeline-range-name>` is not included, the timeline range defaults to [`cover`](/en-US/docs/Web/CSS/Reference/Values/timeline-range-name#cover).

Reviewer: cover only if a view timeline

Author: updated general wording to:

> ...[`cover`](/en-US/docs/Web/CSS/Reference/Values/timeline-range-name#cover) for a view progress timeline source, and [`scroll`](/en-US/docs/Web/CSS/Reference/Values/timeline-range-name#scroll) for a scroll progress timeline source.

> The `timeline-trigger-activation-range-start` and {{cssxref("timeline-trigger-activation-range-end")}} properties can also be set using the {{cssxref("timeline-trigger-activation-range")}} shorthand.

Reviewer: ```suggestion
The `timeline-trigger-activation-range-start`, along with the {{cssxref("timeline-trigger-activation-range-end")}} property, can also be set using the {{cssxref("timeline-trigger-activation-range")}} shorthand.
```
(lingo used in animation-range-*)

Author: updated on all pages

### Specifying multiple range start values

When multiple, comma-separated values are specified in a single `timeline-trigger-activation-range-start` property, they are distributed between the specified {{cssxref("timeline-trigger-name")}} values in the same fashion as other [multiple animation property values are set](/en-US/docs/Web/CSS/Guides/Animations/Using#setting_multiple_animation_property_values).

> For example, if multiple `timeline-trigger-name` values are set, but only a single `timeline-trigger-activation-range-start` value is set, the `timeline-trigger-activation-range-start` will apply to all the `timeline-trigger-name`s. If two `timeline-trigger-activation-range-start` values are set, they will cycle between the `timeline-trigger-name`s until all of them have a `timeline-trigger-activation-range-start` value set. And so on.

Reviewer: linog used elsewhere is:

>> [!NOTE]
> When you specify multiple comma-separated values on an `animation-*` property, they are applied to the animations in the order in which the {{cssxref("animation-name")}}s appear. For situations where the number of animations and `animation-*` property values do not match, see [Setting multiple animation property values](/en-US/docs/Web/CSS/Guides/Animations/Using#setting_multiple_animation_property_values).

Author: I updated the opening bit to match your wording here, in each case.

Author: With a few tweaks so it fits.

Consider these declarations:

```css
timeline-trigger-name: --my-trigger, --my-other-trigger, --another-trigger;
timeline-trigger-activation-range-start:
  cover,
  entry 40%;
```

> In this case, `--my-trigger` will use the `cover` range start and `--my-other-trigger` will use the `entry 40%` range start. As there are three names but only two range ends, the range ends are cycled, so the third trigger name, `--another-trigger`, will use the `cover` range start.

Reviewer: range starts. 

also, not sure if this needs to be explained everywhere, or see the previous comment

Author: I don't understand this comment.

Reviewer: ```suggestion
In this case, as there are three names but only two range ends, the range ends are cycled; with the odd trigger names getting the first `timeline-trigger-activation-range-start` value and the even trigger names getting the second. Here, the first and trigger names, `--my-trigger` and `--another-trigger` respectively, will use the `cover` range start, while  `--my-other-trigger` will use the `entry 40%` range start.
```

Author: I've looked at this and fixed the end/start snafu on a few pages — good spot.

I don't think the wider rewording suggestions are needed. Just seems to be a wordier way to say the same thing, plus I'd have to change it on several pages, so I'm leaving that.

## Formal definition

{{cssinfo}}

## Formal syntax

{{csssyntax}}

## Examples

### Basic usage

> In this example, we show how to create a basic scroll-triggered animation with a custom activation range start value.

Reviewer: ```suggestion
In this example, we inset the starting trigger in a scroll-triggered animation by setting a custom `timeline-trigger-activation-range-start` value.
```
i think this is what this is doing (haven't finished reading yet)

Author: I wasn't sure about this wording, so I decided to go for this wording structure on all pages:

> In this example, we inset the start of a scroll-triggered animation trigger's activation range by setting a custom `timeline-trigger-activation-range-start` value.

#### HTML

> Our markup contains two {{htmlelement("div")}} elements, plus some basic text content to cause the page to scroll. We have hidden the text content for brevity.

Reviewer: ```suggestion
Our markup contains two {{htmlelement("div")}} elements - one is a trigger and the other is the element we will be animating -  plus some basic text content to cause the page to scroll. We have hidden the text content for brevity.
```

Author: I don't see the need to change this (on 10 separate pages), as the text inside the HTML makes it obvious which role each one has. Plus, the element itself is not the trigger; the trigger is created on it.

Reviewer: i always like to not rely on people reading the text nodes in the html

Author: Ok, I've added an update to each page to cover it.

@keyframes rotate {
  from {
    rotate: 0deg;
  }

  to {
    rotate: 360deg;
  }
}
```

We apply the animation to the `.animated` `<div>` for infinite iterations using the {{cssxref("animation")}} shorthand. We then set an {{cssxref("animation-trigger")}} value on it that references a trigger name of `--t` and specifies two {{cssxref("animation-action")}} values — `play` and `pause` — which specify that the animation will play on activation, and pause on deactivation.

```css live-sample___basic-example
div.animated {
  animation: rotate 3s infinite linear;
  animation-trigger: --t play pause;
}
```

The `.trigger` `<div>` element creates a trigger for the animated `<div>` using:

- A {{cssxref("timeline-trigger-name")}} value of `--t`, which is equal to the identifier referenced in the animated `<div>`'s `animation-trigger` property value, associating the two together.
- A {{cssxref("timeline-trigger-source")}} value of [`view()`](/en-US/docs/Web/CSS/Reference/Properties/animation-timeline/view), which sets the timeline trigger as a view progress timeline, and the element providing the timeline trigger as the nearest scrolling ancestor element.

> - A `timeline-trigger-activation-range-start` of `entry 50%`, which sets the trigger's activation range start point to `50%` through the `entry` range (when `50%` of the tracked element has entered the scrollport via the scrollport's end edge). The {{cssxref("timeline-trigger-activation-range-end")}} value defaults to `100%` of the way through the `cover` range, meaning that the animation won't pause until the tracked element has left the scrollport in either direction.

Reviewer: ```suggestion
- A `timeline-trigger-activation-range-start` of `entry 50%`, which sets the trigger's activation range start point to `50%` through the `entry` range (when `50%` of the tracked element has entered the scrollport via the scrollport's end edge).
- When not explicitly set, the {{cssxref("timeline-trigger-activation-range-end")}} value defaults to `normal`, which in this case is `100%` of the way through the `cover` range, meaning the deactivation that the animation will `pause` when the tracked element exits the scrollport's start edge.
```
put these two properties on two different lines.
the content is inaccurate the animation is not active when the trigger drops below 50% of it being in the viewport.

Author: I've made the text changes as suggested, but I'm not putting them into separate bullets, as then the fourth bullet would not fit with the line before the bullets.

```css live-sample___basic-example
div.trigger {
  timeline-trigger-name: --t;
  timeline-trigger-source: view();
  timeline-trigger-activation-range-start: entry 50%;
}
```

```css hidden live-sample___basic-example
@supports not (timeline-trigger-activation-range-start: entry 50%) {
  body::before {
    content: "Your browser does not support the timeline-trigger-activation-range-start property.";
    background-color: wheat;
    padding: 1rem 0;
    text-align: center;
    padding: 1rem 0;

    z-index: 1;
    position: fixed;
    inset: 40% 0 auto;
  }
}
```

#### Result

{{EmbedLiveSample("basic-example", "100%", "240")}}

> Try scrolling the content up and down. The animation plays when `50%` of the trigger `<div>` has entered the scrollport and pauses when the trigger has completely exited the scrollport at the opposite edge.

Reviewer: add what happens when you scroll down.

Author: Now I understand the effects of these properties better, I have rewritten the example descriptions across all the pages so that they are more accurate ;-)
